internal/agent/device/device.go

```diff
@@ -210,6 +210,13 @@ func (a *Agent) syncDeviceSpec(ctx context.Context) {
 			return
 		}
 
+		// Policy may defer update; in all cases, warn and roll back to the previous renderedVersion.
+		if errors.Is(syncErr, errors.ErrUpdatePolicyNotReady) || errors.Is(syncErr, errors.ErrDownloadPolicyNotReady) {
+			a.log.Warnf("Policy not ready for update; Attempting to rollback to previous renderedVersion: %s", current.Version())
```

Comment:
Looking better the bug is around the use of the word `rollback` as it seems like something that would happen in error but in this case its expected. so maybe we can do.

```suggestion
			a.log.Warnf("Retry version: %s: %s", current.Version(), syncErr.Error())
```